apps/dashboard/pages/identity-portal/attendee-event-scan/index.tsx

```diff
@@ -81,76 +87,62 @@ export function Index() {
       .limit(20);
 
     if (res.error) {
-      setResponse(res.error);
-      return;
+      return setResponse(res.error);
     }
 
     const logs = res.data;
-    const mappedLogs = await Promise.all(
-      logs.map(async (log) => {
-        const {
-          data: [{ first_name, last_name }],
-        } = await supabase
-          .getClient()
-          .from('user_profiles')
-          .select()
-          .eq('user_id', log.user_id);
 
+    if (logs.length > 0) {
+      // Get participant names
+      const filter = logs.map((log) => `user_id.eq.${log.user_id}`).join(',');
+      const { data: nameData, error } = await supabase
+        .getClient()
+        .from('user_profiles')
+        .select()
+        .or(filter);
+
+      if (error != null) {
```

Comment:
nit: per JS best practice do `!==` instead of `!=` for strict comparison; without this if Supabase SDK returns a falsy but equal type for error (e.g `undefined`) error would not be reported and might lead to bugs

Reply:
Done!